samples-aspnetcore-3x/self-hosted-login/okta-aspnetcore-mvc-example/Controllers/InteractionCodeController.cs

```diff
@@ -0,0 +1,122 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+using Okta.Idx.Sdk;
+using Okta.Idx.Sdk.Configuration;
+using okta_aspnetcore_mvc_example.Models;
+using okta_aspnetcore_mvc_example.Okta;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace okta_aspnetcore_mvc_example.Controllers
+{
+    public class InteractionCodeController : Controller
+    {
+        private readonly IIdxClient idxClient;
+        private readonly HttpClient httpClient;
+
+        public InteractionCodeController(IIdxClient idxClient)
+        {
+            this.idxClient = idxClient;
+            this.httpClient = new HttpClient();
+        }
+
+        public async Task<IActionResult> Callback()
+        {
+            await GetTokensAsync(HttpContext);
+            return Redirect("/Account/Profile");
+        }
+
+        private async Task GetTokensAsync(HttpContext httpContext)
+        {
+            try
+            {
+                IdxConfiguration idxConfiguration = idxClient.Configuration;
+
+                string state = httpContext.Request.Query["state"];
+                string idxContextJson = httpContext.Session.GetString(state);
+
+                Dictionary<string, string> idxContext = JsonConvert.DeserializeObject<Dictionary<string, string>>(idxContextJson);
+
+                string interactionCode = httpContext.Request.Query["interaction_code"];
+                Uri issuerUri = new Uri(idxConfiguration.Issuer);
+                string domain = issuerUri.Authority;
+                Uri tokenUri = new Uri($"https://{domain}/oauth2/default/v1/token");
```

Comment:
Shouldn't this URL be normalized? This should be relative to the issuer uri and not the domain.

Based on the issuer there could be two cases:

1) root issuer - `https://{yourOktaDomain}`
In this case, the token URL would be `https://{yourOktaDomain}/oauth2/v1/token`

2) default/custom issuer - `https://{yourOktaDomain}/oauth2/default`
In this case, the token URL would be `https://{yourOktaDomain}/oauth2/default/v1/token` 

You will therefore have to construct the token URL based on the issuer here. 

cc @laura-rodriguez

Reply:
Acknowledged, I'll update based on the details you provided.